src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -33,6 +33,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       could be removed by the ILLink, causing a trimmed app to crash. -->
   <PropertyGroup Condition="'$(PublishTrimmed)' == 'true'">
     <StartupHookSupport Condition="'$(StartupHookSupport)' == ''">false</StartupHookSupport>
+    <MetadataUpdaterSupport Condition="'$(MetadataUpdaterSupport)' == ''">false</MetadataUpdaterSupport>
```

Comment:
Can you test that hot reload still works when you `dotnet run` (or `dotnet watch`) a project that has `PublishAot/PublishTrimmed=true` in the .csproj?

I think setting this property is going to break hot reload at debug time.

Reply:
I tried:

```csharp
using System.Runtime.CompilerServices;

while (true)
{
    Do();
    await Task.Delay(500);
}

[MethodImpl(MethodImplOptions.NoInlining)]
static void Do() => Console.WriteLine("Hello, World!");
```

with

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
	  <MetadataUpdaterSupport>false</MetadataUpdaterSupport>
  </PropertyGroup>

</Project>
```

With `dotnet watch` and then tried editing the Hello World string. The metadata does get updated. Did I miss some step?

Reply:
Try using something that uses `MetadataUpdater.IsSupported`:

https://source.dot.net/#System.Private.CoreLib/src/System/Reflection/Metadata/MetadataUpdater.cs,ed8a7ebb4c4b1bc6,references

Like Blazor WASM.

See also - https://github.com/dotnet/runtime/pull/57143#issuecomment-897197278.

Reply:
```csharp
using System.Reflection.Metadata;

Console.WriteLine(MetadataUpdater.IsSupported);
```

This will print False with dotnet run (irrespective of MetadataUpdaterSupport value) and True with dotnet watch (again, irrespective of the property). This still matches my expectations.

Reply:
I didn't know there is C# property. It seems to be marked with `FeatureSwitchDefinition` attribute.
Is `FeatureSwitchDefinition` supposed to do any ILLink substitution ?

Anyway, I just wanted to trim more Mono stuff. I'm happy to revert change in the `src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets` because it should work anyway, right ?

Reply:
The change in `Microsoft.NET.ILLink.targets` looked good to me, we should keep - none of the scenarios @eerhardt asked me to try showed this has any effect so unless there's something I'm missing, this should be good (@eerhardt?).

Reply:
ok, I put it back

Reply:
If you are confident that all the hot reload scenarios still work on projects with `PublishTrimmed=true` in the csproj, then I won't object. I just wanted you to be aware that doing this has broken things in the past.

Reply:
It looks like Maui uses this bool as well - https://source.dot.net/#Microsoft.Maui/HotReload/HotReloadHelper.cs,8b5c57eaeac0896e,references

Reply:
> It looks like Maui uses this bool as well - https://source.dot.net/#Microsoft.Maui/HotReload/HotReloadHelper.cs,8b5c57eaeac0896e,references

The value of this property doesn't change outside of scenarios where trimming executed. We do not read this as an AppContext switch anywhere (https://github.com/search?q=org%3Adotnet+System.Reflection.Metadata.MetadataUpdater.IsSupported&type=code&p=1). This is not a feature switch like the other feature switches in this sense (IsSupported property is true when you run dotnet watch, but it's false for dotnet run or dotnet publish).

The only code I can find is the substitution that _trimming_ will do when it executes.

Reply:
OK, that resolves my concern.